tests/utils/check_warnings.py

```diff
@@ -27,30 +28,37 @@ def check_warnings(file: Path) -> bool:
 
     # find the file where all the known warnings are stored
     warning_file = Path(__file__).parent.parent / "warning_list.txt"
+    extra_warning_file = Path(__file__).parent.parent / "intermittent_warning_list.txt"
 
-    test_warnings = file.read_text().strip().split("\n")
-    ref_warnings = warning_file.read_text().strip().split("\n")
+    received_warnings = escape_ansi(file.read_text()).strip().split("\n")
+    expected_warnings = warning_file.read_text().strip().split("\n")
+    intermittent_warnings = extra_warning_file.read_text().strip().split("\n")
+    # filter out empty warnings (happens on notebooks for some reason)
+    received_warnings = list(filter(len, received_warnings))
 
     print(
         f'Checking build warnings in file: "{file}" and comparing to expected '
-        f'warnings defined in "{warning_file}"\n\n'
+        f'warnings defined in "{warning_file}" and "{extra_warning_file}"\n\n'
     )
 
-    # find all the missing warnings
-    missing_warnings = []
-    for wa in ref_warnings:
-        index = [i for i, twa in enumerate(test_warnings) if wa in twa]
-        if len(index) == 0:
-            missing_warnings += [wa]
-            print(f"{Fore.YELLOW}Warning was not raised: {Fore.RESET}{wa}\n")
-        else:
-            test_warnings.pop(index[0])
-
-    # the remaining one are unexpected
-    for twa in test_warnings:
-        print(f"{Fore.YELLOW}Unexpected warning: {Fore.RESET}{twa}\n")
-
-    return len(missing_warnings) != 0 or len(test_warnings) != 0
+    for exp_w in expected_warnings[::-1] + intermittent_warnings[::-1]:
+        found = False
+        for rec_w in received_warnings:
+            if exp_w in rec_w:
+                received_warnings.remove(rec_w)
+                if exp_w in expected_warnings:
+                    expected_warnings.remove(exp_w)
+                elif exp_w in intermittent_warnings:
```

Comment:
I really like this approach of checking presence but discarding absence. nice trick